client/src/pages/LoginScreen/LoginScreen.tsx

```diff
@@ -0,0 +1,141 @@
+import axios, { AxiosError } from "axios";
+import { UserResponse } from "./LoginScreen.state";
+
+function LoginScreen() {
+  const [error, setError] = useState<string | null>(null);
+  const signIn = useSignIn();
+  const isAuthenticated = useIsAuthenticated();
+  const validateSchema = Yup.object().shape({
+    email: Yup.string()
+      .email("Voer een geldig email adres in.")
+      .required("Voer een geldig email adres in.")
+      .min(8, "Email moet uit minimaal 8 karakters bestaan."),
+    password: Yup.string()
+      .required("Voer een geldig wachtwoord in.")
+      .min(6, "Wachtwoord moet uit minimaal 6 karakters bestaan."),
+  });
+  const formik = useFormik({
+    initialValues: {
+      email: "",
+      password: "",
+    },
+    onSubmit: async (values) => {
+      // Add 'async' keyword to make the function asynchronous
+      if (formik.isValid) {
+        const token: string | undefined = await retrieveToken(); // Await the retrieveToken() function
+        if (!token) {
+          return;
+        }
+        const userData = await retrieveUser(token as string);
+        if (!userData) {
+          return;
+        }
+        const signinres = signIn({
+          auth: {
+            token: token,
+            type: "Bearer",
+          },
+          userState: {
+            id: userData?.id,
+            email: userData?.email,
+            firstName: userData?.firstName,
+            lastName: userData?.lastName,
+          },
+        });
+
+        if (signinres) {
+          window.location.href = "/";
+        }
+      }
+    },
+    validationSchema: validateSchema,
+  });
+  async function retrieveUser(
+    token: string
+  ): Promise<UserResponse | undefined> {
+    try {
+      const response = await axios.get(
+        process.env.REACT_APP_API_URL + "/api/users",
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+      return response.data;
+    } catch (e) {
+      const error = e as AxiosError;
+      console.error(error);
+      if (typeof error.response?.data === "string") {
+        setError(error.response.data);
+      } else {
+        setError("Er is iets misgegaan tijdens het inloggen.");
+      }
+    }
+  }
+  async function retrieveToken(): Promise<string | undefined> {
+    try {
+      const response = await axios.post(
+        process.env.REACT_APP_API_URL + "/api/login",
+        {
+          email: formik.values.email,
+          password: formik.values.password,
+        }
+      );
+      return response.data.token;
+    } catch (e) {
+      const error = e as AxiosError;
+      console.error(error);
+      if (typeof error.response?.data === "string") {
+        setError(error.response.data);
+      } else {
+        setError("Er is iets misgegaan tijdens het inloggen.");
+      }
+    }
+  }
+
+  // If already logged in, redirect to dashboard
+  if (isAuthenticated) {
+    console.log("je moeder");
```

Comment:
Nee die van jouwwwww *insert cakedup pikachu holding uno reverse card*

Reply:
Ehhh die is niet van mij...